2-WebApp-graph-user/2-5-HybridFlow/wwwroot/js/RedeemCode.js

```diff
@@ -0,0 +1,29 @@
+// IIFE to acquire token from auth code sent through server.
+(function () {
+    const code = document.cookie.split(';').reduce((cookies, cookie) => {
+        const [name, value] = cookie.split('=').map(c => c.trim());
+        cookies[name] = value;
+        return cookies;
+    }, {})['Microsoft.Identity.Hybrid.Authentication.Cookie'];
+
+    const scopes = ["user.read"];
+
+    console.log(code);
+
+    if (!!code) {
+        msalInstance.acquireTokenByCode({
+            code,
+            scopes
+        });
+
+        // Remove the authcode from cookies
+        document.cookie = document.cookie.split(';').map(cookie => {
+            const [name, _] = cookie.split('=').map(c => c.trim());
+            if (name === 'Microsoft.Identity.Hybrid.Authentication.Cookie') {
+                return name + '=; Max-Age=-99999999999';
+            }
+            return cookie;
+        }).join(';');
+    }
+})();
+
```

Comment:
This JavaScript file is responsible for converting the auth code received from the server stored in the `Microsoft.Identity.Hybrid.Authentication.Cookie` cookie and exchanging it for an access token to be used client side.

After it's exchanged the cookie is deleted.

This only runs if that cookie is set. After the exchange occurs the token is cached with the `msal.js` library.

Reply:
Suggest adding Jason Nutter to to review this (cookies) and also to review the JS implementation of Hybrid SPA